expression/vectorized.go

```diff
@@ -0,0 +1,215 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expression
+
+import (
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+)
+
+func genVecFromConstExpr(ctx sessionctx.Context, expr Expression, input *chunk.Chunk, result *chunk.Column) error {
```

Comment:
Can we use `FieldType` and `Datum` to replace `Expression`? To make it more like a constant.

Reply:
Each expression has its special logic to handle hybrid type, for example:
https://github.com/pingcap/tidb/blob/master/expression/constant.go#L133, 
https://github.com/pingcap/tidb/blob/master/expression/column.go#L53.

If we change this `expr` to `datum`, the user has to handle these hybrid types again manually before using this function, which is not convenient.

Reply:
PTAL @zz-jason

Reply:
okay